tests/backends/test_kafka.py

```diff
@@ -231,6 +231,90 @@ def test_consumer_stream_processor_shutdown(self) -> None:
+        factory = mock.Mock()
+        factory.create_with_partitions.return_value = strategy
+
+        partition_count = 2
+
+        with self.get_topic(partition_count) as topic, closing(
+            self.get_producer()
+        ) as producer, closing(
+            self.get_consumer(
+                "test_assign_partition_during_pause", enable_end_of_partition=True
+            )
+        ) as consumer_a, closing(
+            self.get_consumer(
+                "test_assign_partition_during_pause", enable_end_of_partition=True
+            )
+        ) as consumer_b:
+            for i in range(partition_count):
+                producer.produce(Partition(topic, i), next(payloads)).result(
+                    timeout=5.0
+                )
+
+            processor_a = StreamProcessor(consumer_a, topic, factory, IMMEDIATE, handle_poll_while_paused=True)
+
+            def wait_until_consumer_pauses(processor: StreamProcessor[Any]) -> None:
+                for _ in range(20):
+                    try:
+                        processor._run_once()
+                    except EndOfPartition:
+                        pass
+
+                    if processor._StreamProcessor__is_paused:  # type:ignore
+                        return
+                print(processor._StreamProcessor__consumer.tell())  # type:ignore
+                raise RuntimeError("processor was not paused")
+
+            # calling _run_once will pause both consumers because of the MessageRejected strategy above
+            wait_until_consumer_pauses(processor_a)
+
+            # consumer A has all the partitions
+            assert len(consumer_a.tell()) == 2
+            assert len(consumer_b.tell()) == 0
+
+            # consumer A has all partitions paused (both from consumer and from
+            # StreamProcessor POV)
+            assert consumer_a.paused()
+            assert processor_a._StreamProcessor__is_paused is True  # type:ignore
+
+            # subscribe with another consumer, now we should have rebalancing in the next few polls
+            processor_b = StreamProcessor(consumer_b, topic, factory, IMMEDIATE, handle_poll_while_paused=True)
+
+            for _ in range(10):
+                try:
+                    processor_a._run_once()
+                except EndOfPartition:
+                    pass
+                try:
+                    processor_b._run_once()
+                except EndOfPartition:
+                    pass
+
+            # balanced
+            assert len(consumer_a.tell()) == 1
+            assert len(consumer_b.tell()) == 1
+
+            # close B, but A has not polled yet, so it only has one partition still
+            consumer_b.close()
+            assert len(consumer_a.tell()) == 1
+
+            for _ in range(20):
+                try:
+                    processor_a._run_once()
+                except EndOfPartition:
+                    pass
+            assert len(consumer_a.tell()) == 2
```

Comment:
Shouldn't we be able to stress the code we changed just by calling `_run_once` on `processor_a` at line 293? That would trigger the rebalance while in paused state without needing the rest of the test. Or am I missing anything ?